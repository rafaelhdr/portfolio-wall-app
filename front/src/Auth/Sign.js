import React, { Component } from 'react';
import { LoginForm } from './LoginForm.js';
import { Logout } from './Logout.js';
import { RegisterForm } from './RegisterForm.js';

class LoginMenu extends Component {

  render() {

    if (this.props.isAuthenticated) {
      return <Logout
        setUser={this.props.setUser}
      />
    }

    return (
      <ul className="nav navbar-nav navbar-right">
        <li className="dropdown">
          <a id="login_register_menu" className="dropdown-toggle" data-toggle="dropdown"><b>Login or Register</b> <span className="caret"></span></a>
          <ul id="login-dp" className="dropdown-menu">
            <li>
              <div className="row">
                <div className="col-md-12">
                  <span>Login</span>
                  <LoginForm
                    setUser={this.props.setUser}
                  />
                  <span>or Register</span>
                  <RegisterForm
                    setUser={this.props.setUser}
                  />
                </div>
              </div>
            </li>
          </ul>
        </li>
      </ul>
    )
  }
}

export { LoginMenu }
