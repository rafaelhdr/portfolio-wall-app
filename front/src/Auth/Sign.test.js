import React from 'react';
import { LoginMenu } from "./Sign.js";
import renderer from 'react-test-renderer';


describe('Components', () => {

  it('render LoginMenu for not authenticated users', () => {
    const tree = renderer
      .create(<LoginMenu isAuthenticated={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render LoginMenu for authenticated users', () => {
    const tree = renderer
      .create(<LoginMenu isAuthenticated={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
