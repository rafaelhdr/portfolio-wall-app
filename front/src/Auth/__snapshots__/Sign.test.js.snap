// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render LoginMenu for authenticated users 1`] = `
<ul
  className="nav navbar-nav navbar-right"
>
  <li>
    <a
      className="hover-pointer"
    >
      Logout
    </a>
  </li>
</ul>
`;

exports[`Components render LoginMenu for not authenticated users 1`] = `
<ul
  className="nav navbar-nav navbar-right"
>
  <li
    className="dropdown"
  >
    <a
      className="dropdown-toggle"
      data-toggle="dropdown"
    >
      <b>
        Login or Register
      </b>
       
      <span
        className="caret"
      />
    </a>
    <ul
      className="dropdown-menu"
      id="login-dp"
    >
      <li>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            <span>
              Login
            </span>
            <form
              acceptCharset="UTF-8"
              className="form"
            >
              <div
                className="form-group"
              >
                <label
                  className="sr-only"
                  htmlFor="login_email"
                >
                  Email address
                </label>
                <input
                  className="form-control"
                  id="login_email"
                  placeholder="Email address"
                  required={true}
                  type="email"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="sr-only"
                  htmlFor="login_password"
                >
                  Password
                </label>
                <input
                  className="form-control"
                  id="login_password"
                  placeholder="Password"
                  required={true}
                  type="password"
                />
              </div>
              <div
                className="form-group"
              >
                <button
                  className="btn btn-primary btn-block"
                  type="submit"
                >
                  Login
                </button>
              </div>
            </form>
            <span>
              or Register
            </span>
            <form
              acceptCharset="UTF-8"
              className="form"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  className="sr-only"
                  htmlFor="username"
                >
                  Username
                </label>
                <input
                  className="form-control"
                  id="username"
                  onChange={[Function]}
                  placeholder="Username"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="sr-only"
                  htmlFor="first_name"
                >
                  First name
                </label>
                <input
                  className="form-control"
                  id="first_name"
                  onChange={[Function]}
                  placeholder="First name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="sr-only"
                  htmlFor="last_name"
                >
                  Last name
                </label>
                <input
                  className="form-control"
                  id="last_name"
                  onChange={[Function]}
                  placeholder="Last name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="sr-only"
                  htmlFor="email"
                >
                  Email address
                </label>
                <input
                  className="form-control"
                  id="email"
                  onChange={[Function]}
                  placeholder="Email address"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="sr-only"
                  htmlFor="password"
                >
                  Password
                </label>
                <input
                  className="form-control"
                  id="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <button
                  className="btn btn-primary btn-block"
                  type="submit"
                >
                  Register
                </button>
              </div>
            </form>
          </div>
        </div>
      </li>
    </ul>
  </li>
</ul>
`;
